src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
+     * the notification center preferences section, and any string passed as a `category` should make sense when
+     * displayed in that context.
+     *
+     * In general, event categories should list all the different types of notifications raised by your app. As a
+     * general guide, if a user may want to have different preferences for some subset of notifications created by your
+     * app, then ensure those notifications have a distinct category.
+     *
+     * For example - given a calendar app, notification categories could be:
+     * - `"Upcoming Events"`: Notification that an event is about to start
+     * - `"Event Start"`: Notification raised when event starts, expiring at the event end time
+     * - `"Event Modified"`: When an event is modified
+     * - `"Event Cancelled"`: When an event is cancelled
+     * - `"Event Response"`: An attendee has responded to an event invitation that you created
+     * - `"Daily Agenda"`: A notification sent each morning with event reminders
+     *
+     * **NOTE:** The user-facing UI that a user would use to manage their preferences is still in progress. This
+     * property has been added in advance of this UI being released, to ensure future compatibility.
      */
-    subtitle?: string;
+    category: string;
+
     /**
-     * URL of the icon to be displayed in the Notification.
+     * URL of the icon to be displayed in the notification.
      */
     icon?: string;
+
     /**
-     * Any custom context data associated with the Notification.
+     * Any custom context data associated with the notification.
      */
     customData?: CustomData;
+
     /**
-     * The timestamp shown on the Notification.  This is presentational only - a future date will not incur a scheduling action.
+     * The timestamp shown on the notification. If omitted, the current date/time will be used.
+     *
+     * This is presentational only - a future date will not incur a scheduling action.
      */
     date?: Date;
+
     /**
-     * Text and icons for up to two Notification action buttons.
+     * A timestamp at which point the notification will be removed from the user's notification center.
+     *
+     * If omitted or `null`, notification will persist until user explicitly actions/clears it.
      */
-    buttons?: OptionButton[];
-}
+    expires?: Date|null;
 
-/**
- * Configuration options for constructing a Button within a Notification.
- */
-export interface OptionButton {
-    title: string;
-    iconUrl?: string;
+    /**
+     * A list of button definitions, to display below the notification text.
+     *
+     * Notifications support up to four buttons, attempting to add more than four will result in the notification
+     * being rejected by the service.
+     */
+    buttons?: ButtonOptions[];
+
+    /**
+     * An action that will fire when the notification is clicked. This only fires on clicks to the notification body,
```

Comment:
"An action result to be passed back to the application inside the notification-action event fired when the notification is clicked"